### PR TITLE
Fix del-vol Jobs created by CSI Controller didn't inherit attributes from CSI Node Pod

### LINKED ISSUE #587: [BUG] JUICEFS_MOUNT_IMAGE does not take effect for Job/delvol
**What happened**:

As documented, the environment `JUICEFS_MOUNT_IMAGE` can use a different image for MountPods, this is OK:

```text
# kubectl get -n kube-system ds/juicefs-csi-node -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="JUICEFS_MOUNT_IMAGE")].value}'
example.com/repo/test:1.0.0
```

Relavent code: 

https://github.com/juicedata/juicefs-csi-driver/blob/14793999fbabbfecb2722524b7312553b2796bba/cmd/controller.go#L69

However, when deleting the Workload and PVC, the created Job (to delete the relevant PV) will not use the image as specified in `JUICEFS_MOUNT_IMAGE`.

```text
# kubectl get job/juicefs-5052b2e4-delvol -n kube-system -ojsonpath='{.spec.template.spec.containers[0].image}'
juicedata/mount:v1.0.3-4.8.3
```

It is also unable to solve this problem by setting `juicefs/mount-image` parameter in StorageClass. Relevant code:

https://github.com/juicedata/juicefs-csi-driver/blob/14793999fbabbfecb2722524b7312553b2796bba/pkg/config/setting.go#L246

After looking around, maybe the reason is in `DeleteVolume()`, `VolCtx` is not passed to `JfsDeleteVol`:

https://github.com/juicedata/juicefs-csi-driver/blob/14793999fbabbfecb2722524b7312553b2796bba/pkg/driver/controller.go#L122

**What you expected to happen**:

The job for delete volumes will also use that image.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS CSI Driver version (which image tag did your CSI Driver use):
- Kubernetes version (e.g. `kubectl version`):
- Object storage (cloud provider and region):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

### PR DESCRIPTION
**Pull request by fungaren**
Fix https://github.com/juicedata/juicefs-csi-driver/issues/587

This PR just works for me, and it might not be the best way to fix the problem. I open this PR is just for a clue, so that anyone can use it as a reference.

### CODE REVIEW

File: cmd/controller.go [hunk 1]
```diff
@@ -69,6 +74,20 @@ func parseControllerConfig() {
 	if mountPodImage := os.Getenv("JUICEFS_MOUNT_IMAGE"); mountPodImage != "" {
 		config.MountImage = mountPodImage
 	}
+
+	k8sclient, err := k8s.NewClient()
+	if err != nil {
+		klog.V(5).Infof("Can't get k8s client: %v", err)
+		os.Exit(0)
+	}
+	ds, err := k8sclient.AppsV1().DaemonSets(config.Namespace).Get(context.TODO(), JUICEFS_CSI_NODE_NAME, metav1.GetOptions{})
```
Comment (zwwhdls) [inline cmd/controller.go]: Can you move Kubernetes client related function in [`pkg/k8sclient/client.go`](https://github.com/juicedata/juicefs-csi-driver/blob/master/pkg/k8sclient/client.go) to keep the structure of the project intact ?

File: deploy/kubernetes/base/resources.yaml [hunk 2]
```diff
@@ -42,7 +42,9 @@ rules:
   - apiGroups: [ "" ]
     resources: [ "endpoints" ]
     verbs: [ "get", "list", "watch", "create", "update", "patch" ]
-
+  - apiGroups: [ "apps" ],
+    resources: [ "daemonsets" ]
+    verbs: [ "get", "list", "watch", "create", "update", "patch" ],
```
Comment (zwwhdls) [inline deploy/kubernetes/base/resources.yaml]: Do we need so many permissions? Or keeping `get` and `list` is just ok?

**Review by zwwhdls — CHANGES_REQUESTED** [review]

File: deploy/kubernetes/base/resources.yaml (hunk 2 above)
Comment (fungaren) [reply deploy/kubernetes/base/resources.yaml]: done

File: cmd/controller.go (hunk 1 above)
Comment (fungaren) [reply cmd/controller.go]: Ok, updated.

File: deploy/kubernetes/base/resources.yaml [hunk 3]
```diff
@@ -42,7 +42,9 @@ rules:
   - apiGroups: [ "" ]
     resources: [ "endpoints" ]
     verbs: [ "get", "list", "watch", "create", "update", "patch" ]
-
+  - apiGroups: [ "apps" ],
+    resources: [ "daemonsets" ]
+    verbs: [ "get", "list" ],
```
Comment (zwwhdls) [inline deploy/kubernetes/base/resources.yaml]: No comma, you can execute `deploy/k8s.yaml` to generate `k8s.yaml` and `k8s_before_v1_18.yaml` from `base/resources.yaml` automatically.

Comment (fungaren) [reply deploy/kubernetes/base/resources.yaml]: Fixed. by the way, I failed to run `make deploy/k8s.yaml` as I am using kustomize v5.0.0, seems the `kustomization.conf` is not compatible with it.

```text
echo "# DO NOT EDIT: generated by 'kustomize build'" > deploy/k8s.yaml
kustomize build deploy/kubernetes/release >> deploy/k8s.yaml
Error: invalid Kustomization: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go struct field Kustomization.patches of type types.Patch
make: *** [deploy/k8s.yaml] Error 1
```

Comment (zwwhdls) [reply deploy/kubernetes/base/resources.yaml]: I have fixed it in https://github.com/juicedata/juicefs-csi-driver/pull/590. Can you merge latest master branch and try again?

Comment (fungaren) [reply deploy/kubernetes/base/resources.yaml]: It works now:

```
$ make deploy/k8s.yaml
echo "# DO NOT EDIT: generated by 'kustomize build'" > deploy/k8s.yaml
kustomize build deploy/kubernetes/release >> deploy/k8s.yaml
cp deploy/k8s.yaml deploy/k8s_before_v1_18.yaml
sed -i.orig 's@storage.k8s.io/v1@storage.k8s.io/v1beta1@g' deploy/k8s_before_v1_18.yaml
```

But no difference to commit, the generated files are not changed.

**Comment by zwwhdls** [comment]
@fungaren In sidecar mode, there is only controller. So, it can not get `juicefs-csi-node` daemonset and crashed when controller started. Can you consider this issue? When daemonset not existed, ignore it.

**Comment by fungaren** [comment]
@zwwhdls Ok, fixed.

**Comment by zwwhdls** [comment]
@fungaren , can you just ignore `sidecar` mode (when config.Webhook==true ) where do not get daemonset from apiserver? Rest LGTM

**Review by Hexilee — APPROVED** [review]
LGTM

**Comment by fungaren** [comment]
@zwwhdls Fixed. In fact I am not familiar to these configs. 😅 By the way, I am considering is it reasonable to just exit with code `0`?

**Comment by zwwhdls** [comment]
@fungaren , CSI Node should be deployed in mount pod mode. `exit 0` is fine when CSI Node not exist.

**Pull request merged**